python/pyspark/sql/connect/plan.py

```diff
@@ -281,9 +281,13 @@ def __init__(
         assert schema is None or isinstance(schema, str)
 
         if options is not None:
+            new_options = {}
             for k, v in options.items():
-                assert isinstance(k, str)
-                assert isinstance(v, str)
+                if v is not None:
+                    assert isinstance(k, str)
+                    assert isinstance(v, str)
+                    new_options[k] = v
+            options = new_options
```

Comment:
Hm, actually Spark Connect doesn't allow `None`. At least it will test Spark Classic path for now so I will just merge this in as it tests at least.

Reply:
hm ok, so not allowing None seems like a separate issue with Spark Connect.
I wonder if there are any options where we intentionally want to use None/null?

Reply:
> I wonder if there are any options where we intentionally want to use None/null?

I am not sure about newly added options but I actually took a look few years ago. Virtually no option that should legitimately support `None` / `null`.